pyNN/brian/standardmodels/electrodes.py

```diff
@@ -161,8 +161,8 @@ def __init__(self, **parameters):
 
     def _generate(self):
         # Note: Brian uses seconds as unit of time
-        temp_num_t = len(numpy.arange(self.start, self.stop + simulator.state.dt * 1e-3, simulator.state.dt * 1e-3))
-        self.times = numpy.array([self.start+(i*simulator.state.dt*1e-3) for i in range(temp_num_t)])
+        temp_num_t = int(round(((self.stop + simulator.state.dt * 1e-3) - self.start) / (simulator.state.dt * 1e-3)))
+        self.times = numpy.array([self.start + (i * simulator.state.dt * 1e-3) for i in range(temp_num_t)])
```

Comment:
This is a slow way to do it.

`self.start + (simulator.state.dt * 1e-3) * numpy.arange(temp_num_t)`

is about 60 times faster for temp_num_t = 10000

Reply:
For information, here is the script I used for the timings:
```python
import timeit

setup = """
import numpy as np
start = 50.0
dt = 0.1
num = 100000
"""

result1 = timeit.timeit('np.array([start + (i * dt * 1e-3) for i in range(num)])', setup, number=1000)
print(result1)

result2 = timeit.timeit('start + (dt * 1e-3) * np.arange(num)', setup, number=1000)
print(result2)

print(result1 / result2)

exec(setup)
a = np.array([start + (i * dt * 1e-3) for i in range(num)])
b = start + (dt * 1e-3) * np.arange(num)
print((a - b).max())
```